### PR TITLE
Generate more structured inputs for fuzzing

### LINKED ISSUE #629: Create structured inputs for fuzzing
This will make it much easier to create test cases from fuzzing failures once we fix them

### PR DESCRIPTION
**Pull request by AdamGS**
Closes #629. Doesn't introduce any functional changes but will make future extension easier IMO.

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -2,19 +2,23 @@
...
-edition = "2021"
+edition = { workspace = true }
 license = { workspace = true }
 
 [package.metadata]
 cargo-fuzz = true
 
 [dependencies]
-libfuzzer-sys = "0.4"
+libfuzzer-sys = { version = "0.4" }
```
Comment (robert3005) [inline fuzz/Cargo.toml]: move this to the top level

File: fuzz/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            .field("action", &self.action)
+            .field("array", &self.array)
+            .finish()
+    }
+}
+
+#[derive()]
+pub enum Action {
+    NoOp,
+    Compress(Box<dyn EncodingCompressor>),
```
Comment (robert3005) [inline fuzz/src/lib.rs]: &'static dyn EncodingCompressor

File: fuzz/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+use vortex_sampling_compressor::compressors::roaring_bool::RoaringBoolCompressor;
+use vortex_sampling_compressor::compressors::roaring_int::RoaringIntCompressor;
+use vortex_sampling_compressor::compressors::runend::DEFAULT_RUN_END_COMPRESSOR;
+use vortex_sampling_compressor::compressors::sparse::SparseCompressor;
+use vortex_sampling_compressor::compressors::zigzag::ZigZagCompressor;
+use vortex_sampling_compressor::compressors::EncodingCompressor;
+
+pub struct FuzzArrayAction {
+    pub array: Array,
+    pub action: Action,
```
Comment (robert3005) [inline fuzz/src/lib.rs]:
```suggestion
    pub action: Vec<Action>,
```
?

Comment (AdamGS) [reply fuzz/src/lib.rs]: Changing that, but building actually chained operations will probably be more work

**Review by robert3005 — APPROVED** [review]

**Pull request merged**